java/code/src/com/redhat/rhn/taskomatic/task/HubReportDbUpdateDriver.java

```diff
@@ -40,54 +39,99 @@
  */
 public class HubReportDbUpdateDriver implements QueueDriver<MgrServerInfo> {
 
+    private static Set<MgrServerInfo> currentMgrServerInfos = Collections.synchronizedSet(new HashSet<>());
     private Logger log;
-    private static AtomicBoolean hasCandidates = new AtomicBoolean(true);
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public void setLogger(Logger loggerIn) {
         this.log = loggerIn;
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public Logger getLogger() {
         return log;
     }
 
-    @Override
-    public List<MgrServerInfo> getCandidates() {
+    public static synchronized Set<MgrServerInfo> getCurrentMgrServerInfos() {
+        return currentMgrServerInfos;
+    }
+
+    /**
+     * Query the existing Mgr Servers and return their infos
+     * @return Set of server info
+     */
+    public Set<MgrServerInfo> getMgrServers() {
         CriteriaBuilder builder = HibernateFactory.getSession().getCriteriaBuilder();
         CriteriaQuery<MgrServerInfo> criteria = builder.createQuery(MgrServerInfo.class);
         criteria.from(MgrServerInfo.class);
         Query<MgrServerInfo> query = HibernateFactory.getSession().createQuery(criteria);
         List<MgrServerInfo> mgrServerInfos = query.list();
-        return mgrServerInfos.stream().filter(info ->
-                Optional.ofNullable(info.getReportDbLastSynced())
-                        .map(time -> time.toInstant().plus(24, ChronoUnit.HOURS).isBefore(Instant.now()))
-                        .orElse(true)
-        ).map(e -> {
-            Hibernate.initialize(e.getReportDbCredentials());
-           return e;
-        }).collect(Collectors.toList());
+        return mgrServerInfos.stream().collect(Collectors.toSet());
```

Comment:
```suggestion
        return new HashSet<>(mgrServerInfos);
```

Better use the constructor instead of iterating the collection with the `stream()`.